src/Matcher/Pattern/TypePattern.php

```diff
@@ -69,10 +78,28 @@ public function matchExpanders($value)
     }
 
     /**
-     * @return null|string
+     * {@inheritdoc}
      */
     public function getError()
     {
         return $this->error;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasExpander(string $expanderName)
+    {
+        foreach ($this->expanders as $expander) {
+            if (!$this->expanderInitializer->hasExpanderDefinition($expanderName)) {
```

Comment:
I'm afraid not. Until `hasExpander` checks if expander exist using it name (simple string - key in `$expanderDefinitions` list) but in `$this->expanders` there is a created objects. I think using expnaderInitializer is better than using some kind of regexp to check if the class name matches with `$expanderName`. These also should be some rules name new expanders. 

I have tried, without success, not to used expanderInitializer. Do you have a better idea?